src/erlcloud_sm.erl

```diff
@@ -0,0 +1,137 @@
+-module(erlcloud_sm).
+-author("joshua@halloapp.com").
+
+-include("erlcloud.hrl").
+-include("erlcloud_aws.hrl").
+
+%%% Library initialization.
+-export([new/2, new/3, new/4]).
+
+%%% API
+-export([
+    get_secret_value/1, get_secret_value/2, get_secret_value/3
+]).
+
+%%%------------------------------------------------------------------------------
+%%% Shared types
+%%%------------------------------------------------------------------------------
+
+-type version_type() :: id | stage.
+
+%%%------------------------------------------------------------------------------
+%%% Library initialization.
+%%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey
+    }.
+
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host
+    }.
+
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host,
+        sm_port = Port
+    }.
+
+%%------------------------------------------------------------------------------
+%% GetSecretValue
+%%------------------------------------------------------------------------------
+%% @doc
+%% SM API:
+%% [https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html]
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec get_secret_value(SecretId :: binary()) -> proplist().
+get_secret_value(SecretId) ->
+    get_secret_value(SecretId, undefined, erlcloud_aws:default_config()).
+
+
+-spec get_secret_value(SecretId :: binary(), Version :: {version_type(), binary()}) -> proplist().
+get_secret_value(SecretId, Version) ->
```

Comment:
So, looking at the API, I think it might be best to make this `Options` like:

```erlang
-type get_secret_value_option() :: {version_id | version_stage, binary()}.
-type get_secret_value_options() :: [get_secret_value_option()].
```

Then you can just map the given `Options` into API parameters, stick `{<<"SecretId">>, SecretID}` on the front of the list, and be done with it.

The other thing that I don't like about `version_type() :: id | stage` is that this introduces nomenclature that isn't used in the AWS API, and for developers, that can become unnatural and confusing, adding another layer of translation to have to keep track of. As above, I really suggest just mapping the optional parameter names exactly as they are, modulo the TitleCase -> snake_case conversion that is conventional for atom representations in Erlang (and this library).

I can see that AWS says `VersionId` and `VersionStage` are mutually exclusive, but I'd advise against baking their constraints into your API. If you introduce the concept of `options` for all optional parameters, the client will be much easier to update and extend in a backwards compatible if/when AWS extends their API.

Other than that issue, I think this looks great. Thanks for the contribution!